src/pages/VideoPlayer/VideoPlayer.jsx

```diff
@@ -21,20 +28,92 @@ const Container = styled(Grid)({
+    const currentLatList = storage.get(WATCH_LATER_LIST);
+    const [isFavorite, setIsFavorite] = useState(video ? currentFavList.find(f => f.id.videoId === video.id.videoId) : [])
+    const [isWatchLater, setIsWatchLater] = useState(video ? currentLatList.find(f => f.id.videoId === video.id.videoId) : [])
     const [related, setRelated] = useState([]);
-    const video = location.state.video;
 
     useEffect(() => {
-        gapi.get('/search', {
-          params: {
-            maxResults: 15,
-            relatedToVideoId: video.id.videoId,
-            type: "video"
-          }
-        }).then((response) =>{
-          setRelated([...response.data.items]);
-        })
-      }, [video]);
+        if (video) {
+            console.log('cambia video')
+            console.log(video.id.videoId)
+            if (from === 'home') {
+                setRelated(relatedResult);
+                // gapi.get('/search', {
+                //     params: {
+                //       maxResults: 15,
+                //       relatedToVideoId: video.id.videoId,
+                //       type: "video"
+                //     }
+                //   }).then((response) =>{
+                //     setRelated([...response.data.items]);
+                //   })
+            }
+
+            if (from === 'favorites') {
+                const favList = storage.get(FAVORITES_LIST);
+                console.log([...favList.filter(f => f.id.videoId !== video.id.videoId)])
+                setRelated([...favList.filter(f => f.id.videoId !== video.id.videoId)])
+            }
+
+            if (from === 'watchLater') {
+                const latList = storage.get(WATCH_LATER_LIST);
+                console.log([...latList.filter(f => f.id.videoId !== video.id.videoId)])
+                setRelated([...latList.filter(f => f.id.videoId !== video.id.videoId)])
+            }
+        }
+
+    }, [video, from]);
+
+    useEffect(() => {
+        if (video) {
+            const favList = storage.get(FAVORITES_LIST);
+            const latList = storage.get(WATCH_LATER_LIST);
+            if (favList.find(f => f.id.videoId === video.id.videoId))
+                setIsFavorite(true)
+            else setIsFavorite(false);
+            if (latList.find(f => f.id.videoId === video.id.videoId))
+                setIsWatchLater(true)
+            else setIsWatchLater(false);
+        }
+
+    }, [video]);
+
+    if (location.state === undefined) {
+        history.push('/notFound')
+        return (
+            <NotFound />
+        )
+    }
+    const addToFavorites = () => {
+        const favList = storage.get(FAVORITES_LIST);
+        storage.set(FAVORITES_LIST, [...favList, video]);
+        setIsFavorite(true);
+    };
+
+    const removeFromFavorites = () => {
+        const favList = storage.get(FAVORITES_LIST);
+        storage.set(FAVORITES_LIST, [...favList.filter(f => f.id.videoId !== video.id.videoId)]);
```

Comment:
there is not need to destructure the array, `filter` returns an array of the items that match the condition

```suggestion
        storage.set(FAVORITES_LIST, favList.filter(f => f.id.videoId !== video.id.videoId));
```